graphix/simulator.py

```diff
@@ -44,14 +44,12 @@ def __init__(self, pattern, backend="statevector", noise_model=None, **kwargs):
         elif backend == "densitymatrix":
             if noise_model is None:
                 self.noise_model = None
-                # no noise: no need to compute probabilities
                 self.backend = DensityMatrixBackend(pattern, **kwargs)
                 warnings.warn(
                     "Simulating using densitymatrix backend with no noise. To add noise to the simulation, give an object of `graphix.noise_models.Noisemodel` to `noise_model` keyword argument."
                 )
             if noise_model is not None:
                 self.set_noise_model(noise_model)
-                # if noise: have to compute the probabilities
                 self.backend = DensityMatrixBackend(pattern, pr_calc=True, **kwargs)
         elif backend in {"tensornetwork", "mps"} and noise_model is None:
```

Comment:
`TensorNetworkBackend` already raises a `TypeError:  __init__() got an unexpected keyword argument 'input_state'` if an `input_state` is given (as tested in 2640845 I just pushed). Is it enough or should we turn this error into a `NotImplementedError`?

Reply:
I personally think that it's good to throw notimplementederror so that it's clear why input state cannot be passed (rather than a bug). Perhaps worth mentioning in changelog that TN won't work with arb input.